lib/index.js

```diff
@@ -88,23 +88,72 @@ function _ensure_synopsis (value) {
 class ValidationError extends Error {
 }
 
+/** */
 class BaseAsset {
     constructor() {
         this._asset_id = new_asset_id();
         this._ekn_tags = [];
         this._tags = [];
     }
 
+    /**
+     * 20-byte hex ID of the document
+     * @type string
+     * @readonly
+     */
     get asset_id() { return this._asset_id; }
 
+    /**
+     * @param {string} value - Title of the document
+     */
     set_title(value) { this._title = value.trim(); }
+    /**
+     * Sets the text that will be shown as the synopsis of a document. (The
+     * synopsis is often shown in the UI on "cards" if there is enough space.)
+     * @param {string} value - Text describing the document
+     */
     set_synopsis(value) { this._synopsis = _ensure_synopsis(value); }
+    /**
+     * Sets an image asset to be the article's thumbnail. (An image asset is
+     * returned e.g. from {@link utildownload_img|util.download_img} or
+     * {@link utildownload_image|util.download_image}.)
+     * @param {ImageAsset} value - Image
+     */
     set_thumbnail(value) { this._thumbnail_asset_id = value.asset_id; }
+    /**
+     * @param {string} value
+     */
     set_canonical_uri(value) { this._canonical_uri = value; }
+    /**
+     * @param {(Date|number|string)} value
+     */
     set_last_modified_date(value) { this._last_modified_date = _ensure_date(value); }
+    /**
+     * Pass the date that the post or article was published. You can pass a
+     * Date object here. If it is not a Date object, then `new Date()` will be
+     * called on it, so you can also pass e.g. a timestamp.
+     * @param {(Date|number|string)} value - The publishing date of this
+     *   document
+     */
     set_date_published(value) { this._date_published = _ensure_date(value); }
+    /**
+     * @param {number} value
+     */
     set_sequence_number(value) { this._sequence_number = value; }
+    /**
+     * @param {string} value
+     */
     set_license(value) { this._license = value; }
+    /**
+     *  Pass in an array of ID strings for "sets" (categories, tags). The ID
```

Comment:
Extra leading whitespace

Reply:
"Set" is the preferred term in the modular framework's API, it was a term picked for that purpose by the designers. I think it would be best to keep it.

Reply:
The modular framework only makes sense in context of everything else but it's not applicable to someone just writing a simple ingester and using the previewer to check their work thus the comment. I don't feel that strongly about it so it's fine if left as-is.